kube-client/src/client/middleware/extra_headers.rs

```diff
@@ -0,0 +1,46 @@
+use std::sync::Arc;
+
+use http::{header::HeaderName, request::Request, HeaderValue};
+use tower::{Layer, Service};
+
+#[derive(Clone)]
+/// Layer that adds a static set of extra headers to each request
+pub struct ExtraHeadersLayer {
```

Comment:
I thought we can use [`Option<SetRequestHeaderLayer>`](https://docs.rs/tower-http/latest/tower_http/set_header/struct.SetRequestHeaderLayer.html) for user and groups. But I guess this is necessary because `impersonate-group: a,b,c` is not supported :(

- https://github.com/kubernetes/kubernetes/issues/94683
- https://github.com/kubernetes/kubernetes/issues/82468

Reply:
Yeah.. SRHL also means that we need to encode the exact number of headers into the type signature (or do a bunch of newtyping to hide it).